src/main/java/streambms/IInputType.java

```diff
@@ -0,0 +1,7 @@
+package streambms;
+
+public interface IInputType
+{
+  String getInput(); // input type could reading required number of samples from console or

```

Comment:
This interface reads only one line, it looks like. Make it clear by calling it `getOneLineOfInput`. Then it's clear that it needs to be called in a loop inside main()